programs/clearing_house/src/math/fees.rs

```diff
@@ -27,34 +37,65 @@ pub fn calculate_fee_for_order_fulfill_against_amm(
         let fee_to_market = fee.checked_sub(filler_reward).ok_or_else(math_error!())?;
         let user_fee = 0_u128;
 
-        Ok((user_fee, fee_to_market, filler_reward))
+        Ok(FillFees {
+            user_fee,
+            maker_rebate: 0,
+            fee_to_market,
+            filler_reward,
+            referee_discount: 0,
+            referrer_reward: 0,
+        })
     } else {
         let fee = quote_asset_amount
             .checked_mul(fee_structure.fee_numerator)
             .ok_or_else(math_error!())?
             .checked_div(fee_structure.fee_denominator)
             .ok_or_else(math_error!())?;
 
-        let user_fee = fee;
+        let (referrer_reward, referee_discount) = if reward_referrer {
```

Comment:
it might be good to include the user stats and the ref stats in the function, so we can have stats based fee logic/discounts design wrapped up too?

Reply:
i was thinking i'd add that once it is clear what the requirement is